cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/ETLTPFSTest.java

```diff
@@ -0,0 +1,141 @@
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.api.dataset.lib.TimePartitionedFileSet;
+import co.cask.cdap.proto.AdapterConfig;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.template.etl.batch.config.ETLBatchConfig;
+import co.cask.cdap.template.etl.common.ETLStage;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.test.AdapterManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.tephra.Transaction;
+import co.cask.tephra.TransactionAware;
+import co.cask.tephra.TransactionManager;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.twill.filesystem.Location;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class ETLTPFSTest extends BaseETLBatchTest {
+
+  private static final Gson GSON = new Gson();
+  private static Schema EVENT_SCHEMA_WITHOUT_HEADER;
+  @Test
+  public void testAvroSourceConversionToAvroSink() throws Exception {
+
+    Schema schema = Schema.recordOf(
+      "record",
+      Schema.Field.of("int", Schema.of(Schema.Type.INT)));
+    EVENT_SCHEMA_WITHOUT_HEADER = schema;
+
+    org.apache.avro.Schema avroSchema = convertSchema(schema);
+
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("int", Integer.MAX_VALUE)
+      .build();
+
+    String filesetName = "tpfs";
+    DatasetAdmin datasetAdmin = addDatasetInstance("timePartitionedFileSet", filesetName, FileSetProperties.builder()
+      .setInputFormat(AvroKeyInputFormat.class)
+      .setOutputFormat(AvroKeyOutputFormat.class)
+      .setInputProperty("schema", avroSchema.toString())
+      .setOutputProperty("schema", avroSchema.toString())
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", (avroSchema.toString()))
+      .build());
+    DataSetManager<TimePartitionedFileSet> fileSetManager = getDataset(filesetName);
+    TimePartitionedFileSet tpfs = fileSetManager.get();
+
+    TransactionManager txService = getTxService();
+    Transaction tx1 = txService.startShort(100);
+    TransactionAware txTpfs = (TransactionAware) tpfs;
+    txTpfs.startTx(tx1);
+
+    fileSetManager.get().addPartition(System.currentTimeMillis(), "directory", ImmutableMap.of("key1", "value1"));
```

Comment:
its possible, though unlikely, that the time on line and the next line will cross a partition boundary. For example, the minute becomes different.  It is better to get the time once, then use that value in both of these calls.